File: scripts/regen_og_pngs.js
#!/usr/bin/env node
/**
 * Batch-convert all per-skill OG SVGs to PNG using sharp (librsvg bundled).
 * Works on Windows, macOS, Linux without any system-level Cairo install.
 *
 * Usage (from repo root):
 *   npm install --no-save sharp
 *   node scripts/regen_og_pngs.js
 */
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const OG_DIR = 'docs/og';
const SKIP = new Set(['social-preview.svg']);
const W = 1200, H = 630;

function rglob(dir) {
  const out = [];
  for (const f of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, f.name);
    if (f.isDirectory()) out.push(...rglob(full));
    else if (f.name.endsWith('.svg') && !SKIP.has(f.name)) out.push(full);
  }
  return out;
}

(async () => {
  if (!fs.existsSync(OG_DIR)) {
    console.error('ERROR: docs/og/ not found. Run from repo root.');
    process.exit(1);
  }
  const svgs = rglob(OG_DIR);
  if (!svgs.length) { console.log('No SVGs found.'); return; }
  console.log(`Converting ${svgs.length} SVGs → PNGs (${W}×${H})...`);
  let ok = 0, errors = [];
  for (const svg of svgs) {
    const png = svg.replace(/\.svg$/, '.png');
    try {
      await sharp(svg).resize(W, H).png().toFile(png);
      console.log(`  PNG: ${png}`);
      ok++;
    } catch (e) {
      errors.push({ svg, msg: e.message });
      console.error(`  ERR: ${svg} — ${e.message}`);
    }
  }
  console.log(`\nGenerated ${ok}/${svgs.length} PNG(s).`);
  if (errors.length) process.exit(1);
})();

File: scripts/regen_og_pngs.py
#!/usr/bin/env python3
"""Batch-convert all per-skill OG SVGs to PNG using cairosvg (sequential).

Run from the repo root:
    pip install cairosvg
    python scripts/regen_og_pngs.py

Skips social-preview.svg (different 1280x640 dimensions).
"""
import sys
from pathlib import Path

OG_W = 1200
OG_H = 630
SKIP = {"social-preview.svg"}


def main() -> None:
    try:
        import cairosvg
    except ImportError:
        print("ERROR: cairosvg not installed. Run: pip install cairosvg", file=sys.stderr)
        sys.exit(1)

    og_dir = Path("docs/og")
    if not og_dir.exists():
        print(f"ERROR: {og_dir} not found. Run from repo root.", file=sys.stderr)
        sys.exit(1)

    svgs = sorted(p for p in og_dir.rglob("*.svg") if p.name not in SKIP)
    if not svgs:
        print("No SVGs found in docs/og/")
        return

    png_count = 0
    errors = []
    for svg_path in svgs:
        png_path = svg_path.with_suffix(".png")
        try:
            cairosvg.svg2png(
                url=str(svg_path.resolve()),
                write_to=str(png_path),
                output_width=OG_W,
                output_height=OG_H,
            )
            print(f"  PNG: {png_path.relative_to(og_dir.parent)}")
            png_count += 1
        except Exception as e:
            errors.append((svg_path, e))
            print(f"  ERR: {svg_path} — {e}", file=sys.stderr)

    print(f"\nGenerated {png_count}/{len(svgs)} PNG(s).")
    if errors:
        print(f"{len(errors)} error(s) — see above.", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()
